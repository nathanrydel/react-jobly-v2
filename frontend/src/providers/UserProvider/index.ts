export * from './UserProvider.types';
export * from './UserProvider.context';
export * from './UserProvider';
