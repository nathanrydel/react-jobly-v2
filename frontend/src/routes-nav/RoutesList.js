import React, { useContext } from 'react';
import { Routes, Route, Navigate } from "react-router-dom";

import { AppRoutes } from '~/constants';
import { Homepage } from '~/pages/homepage';

import CompanyList from "../companies/CompanyList";
import JobList from "../jobs/JobList";
import CompanyDetail from "../companies/CompanyDetail";
import LoginForm from "../auth/LoginForm";
import ProfileForm from "../profiles/ProfileForm";
import SignupForm from "../auth/SignupForm";
import { UserProviderContext } from '~/providers';

/** Site-wide routes.
 *
 * Parts of site should only be visitable when logged in.
 *
 * Visiting a non-existent route navigates to the homepage.
 */
function RoutesList() {

  const { currentUser } = useContext(UserProviderContext);

  console.debug(
    "Routes",
    `login=${typeof login}`,
    `register=${typeof register}`,
  );

  return (
    <div className="pt-5">
      <Routes>
        {!currentUser &&
          <>
            <Route path="/login" element={<LoginForm />} />
            <Route path="/signup" element={<SignupForm />} />
          </>
        }

        <Route path={AppRoutes.Homepage} element={<Homepage />} />

        {currentUser &&
          <>
            <Route path="/companies" element={<CompanyList />} />
            <Route path="/jobs" element={<JobList />} />
            <Route path="/companies/:handle" element={<CompanyDetail />} />
            <Route path="/profile" element={<ProfileForm />} />
          </>
        }

        <Route path="*" element={<Navigate to="/" />} />
      </Routes>
    </div>
  );
}

export default RoutesList;
