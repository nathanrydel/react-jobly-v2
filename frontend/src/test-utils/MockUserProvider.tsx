import React, { PropsWithChildren } from "react";
import { UserContext, UserProviderContext } from "~/providers";

interface MockUserProviderProps extends PropsWithChildren<{}> {
  currentUser: UserContext['currentUser'];
}

export const MockUserProvider: React.FC<MockUserProviderProps> = ({ children, currentUser }) => {
  const mockContextValue: UserContext = {
    currentUser,
    setCurrentUser: jest.fn(),
    hasAppliedToJob: jest.fn(),
    applyToJob: jest.fn(),
    logout: jest.fn(),
    login: jest.fn(),
    signup: jest.fn(),
    userLoaded: true,
  };

  return (
    <UserProviderContext.Provider value={mockContextValue}>
      {children}
    </UserProviderContext.Provider>
  );
}