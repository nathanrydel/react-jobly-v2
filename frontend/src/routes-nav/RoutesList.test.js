import React from "react";
import { render } from "@testing-library/react";
import RoutesList from "./RoutesList";
import { MemoryRouter } from "react-router";
import { UserProvider } from '~/providers';
import { MockUserProvider } from '~/test-utils/MockUserProvider';

// Define the demoUser object
const demoUser = {
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  password: "testpassword",
  email: "test@test.net",
  isAdmin: false,
  applications: [],
};

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <RoutesList />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot with demo user", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <MockUserProvider currentUser={demoUser}>
          <RoutesList />
        </MockUserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
