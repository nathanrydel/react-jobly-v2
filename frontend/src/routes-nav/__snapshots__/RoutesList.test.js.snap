// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with demo user 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 fw-bold"
        >
          Jobly
        </h1>
        <p
          class="lead"
        >
          All the jobs in one, convenient place.
        </p>
        <h2>
          Welcome Back, testfirst!
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
